numba/core/typed_passes.py

```diff
@@ -669,3 +679,86 @@ def __init__(self):
     def run_pass(self, state):
         dead_code_elimination(state.func_ir, state.typemap)
         return True
+
+
+@register_pass(mutates_CFG=False, analysis_only=False)
+class PreLowerStripPhis(FunctionPass):
+    """Remove phi nodes (ir.Expr.phi) introduced by SSA.
+
+    This is needed before Lowering because the phi node in Numba IR does not
+    matches the semantic of phi node of LLVM IR. In Numba IR, phi node may
+    expand into multiple LLVM instructions.
+    """
+
+    _name = "strip_phis"
+
+    def __init__(self):
+        FunctionPass.__init__(self)
+
+    def run_pass(self, state):
+        state.func_ir = self._strip_phi_nodes(state.func_ir)
+        state.func_ir._definitions = build_definitions(state.func_ir.blocks)
+        # Rerun postprocessor to update metadata
+        post_proc = postproc.PostProcessor(state.func_ir)
+        post_proc.run(emit_dels=False)
+
+        if (state.func_ir.generator_info is not None
+                and state.typemap is not None):
```

Comment:
Is there a situation where there's generator info and no typemap ?

Reply:
objectmode generator